gnome-base/nautilus: Bump to 46.2, gnome-extra/sushi: Bump to 46.0, dev-libs/glib, dev-util/glib-utils, dev-util/gdbus-codegen: Bump to 2.80.4, dev-libs/gobject-introspection, dev-libs/gobject-introspection-common: Bump to 1.80.1

flint2 opened the pull request:
- dev-libs/gobject-introspection: Bump to 1.80.1
- dev-libs/gobject-introspection-common: Bump to 1.80.1
- gnome-extra/sushi: Bump to 46.0
- dev-libs/glib: Bump to 2.80.2
- dev-util/glib-utils: Bump to 2.80.2
- dev-util/gdbus-codegen: Bump to 2.80.2
- gnome-base/nautilus: Bump to 46.2

<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

cfuga commented on gnome-base/nautilus/nautilus-46.2.ebuild:
Please, remove the dependency on gnome-extra/nautilus-sendto. It's no longer required since version 43.beta.

[https://gitlab.gnome.org/GNOME/nautilus/-/blob/main/NEWS](https://gitlab.gnome.org/GNOME/nautilus/-/blob/main/NEWS)
[https://gitlab.gnome.org/GNOME/nautilus/-/issues/928](https://gitlab.gnome.org/GNOME/nautilus/-/issues/928)
[https://gitlab.gnome.org/GNOME/nautilus/-/issues/2126](https://gitlab.gnome.org/GNOME/nautilus/-/issues/2126)

flint2 replied:
Done in https://github.com/gentoo/gentoo/commit/2ff5a86c3189a493048ae0d26f6595001f72c48b

mm1ke commented on dev-libs/glib/glib-2.80.2.ebuild:
```diff
@@ -0,0 +1,319 @@
+	addpredict /usr/b
+
+	# Related test is a bit nitpicking
+	mkdir "$G_DBUS_COOKIE_SHA1_KEYRING_DIR"
+	chmod 0700 "$G_DBUS_COOKIE_SHA1_KEYRING_DIR"
```
missing `|| die` on both

flint2 replied:
Added in https://github.com/gentoo/gentoo/commit/bf185f660ab4867fa0aae4d60995ac8bbeeb1631

mm1ke commented on dev-libs/glib/glib-2.80.2.ebuild:
```diff
@@ -0,0 +1,319 @@
+			rm -f "${EROOT}"/usr/$(get_libdir)/gio/modules/giomodule.cache
+		}
+		multilib_foreach_abi multilib_pkg_postrm
+		rm -f "${EROOT}"/usr/share/glib-2.0/schemas/gschemas.compiled
```
missing `|| die`

flint2 replied:
Changed in https://github.com/gentoo/gentoo/commit/fc1eeab6c1eca0cc42d8ee062f3761cbc9ddf671

leio replied:
These were not missing, because the die is not necessary, because it's a force removal. Doesn't hurt though.

flint2 commented on dev-libs/glib/glib-2.80.2.ebuild:
```diff
@@ -0,0 +1,319 @@
+
+	if [[ -z ${REPLACED_BY_VERSION} ]]; then
+		multilib_pkg_postrm() {
+			rm -f "${EROOT}"/usr/$(get_libdir)/gio/modules/giomodule.cache
```
Changed in https://github.com/gentoo/gentoo/commit/fc1eeab6c1eca0cc42d8ee062f3761cbc9ddf671

l33tlinuxh4x0r commented:
glib 1.80.1 and gobject-introspection 1.80.1 are required by eachother upstream they say to turn off introspection build and then turn it back on and build again. However there is no introspection use flag.

leio commented:
How are you handling initial install here with the circulars?

pacho2 commented:
Hi!

I have been reviewing the issue and it seems that we will need to live for some release cycles with this circular problem (https://gitlab.gnome.org/GNOME/glib/-/issues/2616). In that case, we will need to have proper "introspection" USE flag handling in glib-2.80 ebuilds to allow the users to build glib without introspection at first, build gobject-introspection, and finally rebuild glib with "introspection".

I wonder about what will be better for Releng team, as they will hit the issue when generating desktop stage3 tarballs. I see that they seem to rely on the "build" USE flag for that (sys-apps/util-linux could be an example). In this case, the "build" USE flag should disable introspection support (as it is being done in the other distributions and their "bootstrap" alike flags to workaround this circular dep issue).

Regards

flint2 commented:
Added +introspection IUSE for glib in https://github.com/gentoo/gentoo/pull/36834/commits/eca89449885b7221d7387c007f04ec6153b97912

This will keep circular dependency and warn users to first emerge glib with -introspection
After that USE can be changed to introspection again
```
 # emerge -av '=dev-libs/glib-2.80.4'

These are the packages that would be merged, in order:

Calculating dependencies... done!
Dependency resolution took 41.55 s (backtrack: 1/20).



[nomerge       ] dev-libs/glib-2.80.4:2::gentoo  USE="dbus elf introspection (mime) xattr -debug -doc (-selinux) -static-libs -sysprof -systemtap -test -utils" ABI_X86="(64) -32 (-x32)" 
[ebuild  N     ]  dev-libs/gobject-introspection-1.80.1::gentoo  USE="-doctool -gtk-doc -test" PYTHON_SINGLE_TARGET="python3_12 -python3_10 -python3_11" 0 KiB
[ebuild  N     ]   dev-libs/glib-2.80.4:2::gentoo  USE="dbus elf introspection (mime) xattr -debug -doc (-selinux) -static-libs -sysprof -systemtap -test -utils" ABI_X86="(64) -32 (-x32)" 0 KiB

Total: 2 packages (2 new), Size of downloads: 0 KiB

 * Error: circular dependencies:

(dev-libs/gobject-introspection-1.80.1:0/0::gentoo, ebuild scheduled for merge) depends on
 (dev-libs/glib-2.80.4:2/2::gentoo, ebuild scheduled for merge) (buildtime)
  (dev-libs/gobject-introspection-1.80.1:0/0::gentoo, ebuild scheduled for merge) (buildtime)

It might be possible to break this cycle
by applying the following change:
- dev-libs/glib-2.80.4 (Change USE: -introspection)

Note that this change can be reverted, once the package has been installed.

```
Also changed metadata to state that introspection require gobject-introspection
```
 + - introspection : Enable generating introspection data (requires dev-libs/gobject-introspection )
```

Not sure if more messages or IUSE flags are needed.

pacho2 commented:
@mattst88 Hi Matt. I see you are also a member of RelEng team. Do you think that you will need some "special handling" (USE build) for generating the stage3 tarballs?

Thanks a lot

mattst88 commented:
Since stage1 is built with `USE=-*`, we might be okay without `IUSE=build` -- but that requires that `glib` is in stage1 which I'm not sure about. Let me check on that.

Otherwise, yeah, we'd probably need `IUSE=build`.

cc: @gentoo/releng

pacho2 commented on dev-libs/gobject-introspection-common/gobject-introspection-common-1.80.1.ebuild:
```diff
@@ -0,0 +1,30 @@
+inherit gnome.org
+
+DESCRIPTION="Build infrastructure for GObject Introspection"
+HOMEPAGE="https://wiki.gnome.org/Projects/GObjectIntrospection"
```
Now it is https://gi.readthedocs.io/

flint2 replied:
Done in https://github.com/gentoo/gentoo/pull/36834/commits/3e58ece98763be422d5cde5e907ccd94776d35af

pacho2 commented on gnome-base/nautilus/nautilus-46.2.ebuild:
```diff
@@ -0,0 +1,121 @@
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc ~ppc64 ~riscv ~x86"
+
+IUSE="+cloudproviders gnome +gstreamer gtk-doc +introspection +previewer selinux"
+REQUIRED_USE="gtk-doc? ( introspection )"
```
As explained at 
https://gitlab.gnome.org/GNOME/nautilus/-/issues/3150

We should rename gtk-doc USE to doc and handle docs normally

This should also Close: https://bugs.gentoo.org/911997

flint2 replied:
Done in https://github.com/gentoo/gentoo/pull/36834/commits/78213b281a9371b220b4ec0969107c89f1c770ec

pacho2 commented on dev-libs/glib/glib-2.80.4.ebuild:
```diff
@@ -0,0 +1,320 @@
+
+EAPI=8
+PYTHON_REQ_USE="xml(+)"
+PYTHON_COMPAT=( python3_{11..13} )
```
Please keep 3.10 support to avoid breaking reverse deps:
https://bugs.gentoo.org/936437

flint2 replied:
Done in https://github.com/gentoo/gentoo/pull/36834/commits/78213b281a9371b220b4ec0969107c89f1c770ec

pacho2 commented on dev-libs/glib/glib-2.80.4.ebuild:
```diff
@@ -0,0 +1,320 @@
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~arm64-macos ~ppc-macos ~x64-macos ~x64-solaris"
+IUSE="dbus debug +elf doc +introspection +mime selinux static-libs sysprof systemtap test utils xattr"
+RESTRICT="!test? ( test )"
+#REQUIRED_USE="gtk-doc? ( test )" # Bug #777636
```
Please drop this if it is not needed anymore

flint2 replied:
Done in https://github.com/gentoo/gentoo/pull/36834/commits/83ead2bc3ad7a07b985e34ed6702a0f8a87dbc14

pacho2 commented on dev-libs/glib/glib-2.80.4.ebuild:
```diff
@@ -0,0 +1,320 @@
+	mkdir "$G_DBUS_COOKIE_SHA1_KEYRING_DIR" || die
+	chmod 0700 "$G_DBUS_COOKIE_SHA1_KEYRING_DIR" || die
+
+	meson_src_test --timeout-multiplier 2 --no-suite flaky
```
Why timeout was reduced from 20 to 2? Maybe it could break on slower arches :/

flint2 replied:
Typo, corrected in https://github.com/gentoo/gentoo/pull/36834/commits/83ead2bc3ad7a07b985e34ed6702a0f8a87dbc14

pacho2 commented on dev-util/gdbus-codegen/gdbus-codegen-2.80.4.ebuild:
```diff
@@ -0,0 +1,79 @@
+
+EAPI=8
+GNOME_ORG_MODULE="glib"
+PYTHON_COMPAT=( python3_{10..12} )
```
Maybe this could be updated to 10..13

flint2 replied:
Done in https://github.com/gentoo/gentoo/pull/36834/commits/78e3aa629d8fb95c29ff0603a383e4e123b1dba9

pacho2 commented on dev-util/gdbus-codegen/gdbus-codegen-2.80.4.ebuild:
```diff
@@ -0,0 +1,79 @@
+
+src_compile() {
+	distutils-r1_src_compile
+	#do_xsltproc_command \
```
Is this expected? At the end, it won't install docs

flint2 replied:
Changed do_xsltproc_command  to rst2man in https://github.com/gentoo/gentoo/pull/36834/commits/05f5673ced3053b44a074d69189d78606302d21b

pacho2 commented on dev-util/gdbus-codegen/gdbus-codegen-2.80.4.ebuild:
```diff
@@ -0,0 +1,79 @@
+
+python_install_all() {
+	distutils-r1_python_install_all # no-op, but prevents QA warning
+	#doman "${WORKDIR}/glib-${PV}/docs/reference/gio/gdbus-codegen.1"
```
The same here, this will cause the man page to not be installed at the end.

Thanks!

flint2 replied:
Done in https://github.com/gentoo/gentoo/commit/05f5673ced3053b44a074d69189d78606302d21b

flint2 commented on dev-libs/gobject-introspection/gobject-introspection-1.80.1.ebuild:
```diff
@@ -0,0 +1,80 @@
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
```
Done in https://github.com/gentoo/gentoo/pull/36834/commits/4b5833928668c528880305048738ee9fa8507e6f

flint2 commented on dev-libs/gobject-introspection/gobject-introspection-1.80.1.ebuild:
```diff
@@ -0,0 +1,80 @@
+inherit gnome.org meson python-single-r1 xdg
+
+DESCRIPTION="Introspection system for GObject-based libraries"
+HOMEPAGE="https://wiki.gnome.org/Projects/GObjectIntrospection"
```
Done in https://github.com/gentoo/gentoo/pull/36834/commits/4b5833928668c528880305048738ee9fa8507e6f

eli-schwartz commented on dev-libs/gobject-introspection/gobject-introspection-1.80.1.ebuild:
```diff
@@ -0,0 +1,80 @@
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
```
This appears to have some really weird gaps. The commit for this package was authored in January, but the PR was opened in *May*. It is missing crucial requirements for py3.12 support that were committed in February: https://github.com/gentoo/gentoo/pull/34945

These long-running PRs that don't get rebased onto newer ebuilds are starting to become a problem, IMHO.
